### ISSUE TITLE
Allow @type { import ... } to define classes

### ISSUE THREAD

**Issue by SimonMeskens** [Microsoft/TypeScript#24467]
TS version: 2.9+ (Relevant issues: https://github.com/Microsoft/TypeScript/issues/14377#issuecomment-379935736, https://github.com/Microsoft/TypeScript/issues/14844, )

test.d.ts
```ts
export interface TestProperties {
    bar: string;
    baz: number;
}

export class Test {
    foo(param: TestProperties);
}
```

test.js
```ts
/**
 * @type { import("./test").Test }
 */
export class Test {
    foo({ bar, baz }) {
        bar = 7;
    }
}
```

Expected behavior: Throw an error when trying to assign 7 to bar
Actual behavior: no error

If I make this change, it works:

test.js
```ts
/**
 * @type { import("./test").Test }
 */
export class Test {
    /**
     * @param { import("./test").TestProperties } param
     */
    foo({ bar, baz }) {
        bar = 7;
    }
}
```

Correct behavior: `Type '7' is not assignable to type 'string'.`

Giving `Test` the correct type definition should also give `foo` the correct definition. Ideally, we wouldn't even need to do either and we should be able to just do:

```ts
/** @typedef { import("./test") } Test */
```

Or just be able to have TS pick up on the fact that we defined a .d.ts file for our .js file

**Comment by SimonMeskens** [comment]
@bennypowers
